src/content/reference/react/index.md

```diff
@@ -10,6 +10,14 @@ This section provides detailed reference documentation for working with React. F
 
 The React reference documentation is broken down into functional subsections:
 
+## Rules of React {/*rules-of-react*/}
+
+React is a programming paradigm for declarative user interfaces, in which developers describe what should be rendered given the current state and the framework is responsible for updating the UI. Just as different programming languages have their own ways of expressing concepts, React has its own idioms — or rules — for how to express patterns in a way that is easy to understand and yields high-quality applications.
```

Comment:
> Just as different programming languages have their own ways of expressing concepts,

This isn't superfluous, it's an explicit comparison of React to programming languages, which we haven't typically made the case for before. I think it's important to draw this new comparison because it makes the React programming model make sense. I will shorten this section here though because it's just the top level index page, and keep it in the actual Rules of React index.